lib/models/ObjectMD.ts

```diff
@@ -649,10 +650,24 @@ export default class ObjectMD {
      * Set metadata transition in progress value
      *
      * @param inProgress - True if transition is in progress, false otherwise
+     * @param transitionTime - Date when the transition started
      * @return itself
      */
-    setTransitionInProgress(inProgress: boolean) {
+    setTransitionInProgress(inProgress: false): this
+    setTransitionInProgress(inProgress: true, transitionTime: Date|string|number): this
+    setTransitionInProgress(inProgress: boolean, transitionTime?: Date|string|number) {
```

Comment:
At compile time, (I think) overloaded function declarations do not raise an error.
Hence, do we need to handle the case where `inProgress` is `true` and `transitionTime` is `undefined`?

Reply:
As we can see on [Typescript playground](https://www.typescriptlang.org/play?#code/PTAEHUFMBsGMHsC2lQBd5oBYoCoE8AHSAZVgCcBLA1UABWgEM8BzM+AVwDsATAGiwoBnUENANQAd0gAjQRVSQAUCEmYKsTKGYUAbpGF4OY0BoadYKdJMoL+gzAzIoz3UNEiPOofEVKVqAHSKymAAmkYI7NCuqGqcANag8ABmIjQUXrFOKBJMggBcISGgoAC0oACCbvCwDKgU8JkY7p7ehCTkVDQS2E6gnPCxGcwmZqDSTgzxxWWVoASMFmgYkAAeRJTInN3ymj4d-jSCeNsMq-wuoPaOltigAKoASgAywhK7SbGQZIIz5VWCFzSeCrZagNYbChbHaxUDcCjJZLfSDbExIAgUdxkUBIursJzCFJtXydajBBCcQQ0MwAUVWDEQC0gADVHBQGNJ3KAALygABEAAkYNAMOB4GRonzFBTBPB3AERcwABS0+mM9ysygc9wASnJjEEwhw+hoAG9FCUSgB9bh1Bj5MScPDBFQ4ZoeMheRDi5zA9g0L5uMzMdgMZiQfiwaDqRIc+B6ERePl0hlMwR8pLYvngByoADkwgAcpAJHygioAPJfMjvQQRrSQI6oRwKVzSPCgJzenTDLD6FAIRDQ4SXQMScXREQGDjYhZMVgcHhBOuoHBkMxyeqNACSnFobFY+kESoy+-gh8NDuSDGgdZ1DqGvxXa438ganF3Z4vx9PB4JBTQMh2HrVB10pN9GhwKFIAdAAROpIAAHypShOGYRDOHYRBpG+e8BCfRsX3ArcPz3P8jxPMjz3-B1gTlDxOH4UDXxIqDkAAfjghDkNA4YMKwnCyB1UBzUtfCAhtO0AG081WUoGQAL1KQRamgUpmOI99SgyUoCHIw08wAXR5RMv3-ABuC1LQRUAlQAQl-aij1ARDENAOyNM3d82MgYTRLEy1uBgRtbiECTbWbGS5MU5TVPUsCvMadToKMyyxIAX3BW8UH8gKRFSJVUHaIlPIgzgfJ5bleTzTDsO+PM-KsvKxNK1joJMzgS1AeCFEKhKyp8nU0ua9KmoCmy+pY7z2oyKkzAsIket8kSxuawCpsg9reVa6bkACdBt2ICtiF4tClSG1bLVGtbH3C6TZPkxAlJUm94o2zhkuQIyTJ2zbkGGkprstJxUHxTI1EESzRtG6VGipNAOq640qXOxRUACZ9+pIz99OPa9sr1dHMfenGnMNPrgP4TqJG6hDzsJjHCKx99Se-Cn6wARgAdgABg5gAWABmABODmBYZ4nNJ3Ki2dAyn+Q54XedKPnSgAJh5nAOYANnyQWufyRWAjF-mAC0+T1NHGdXZnpbMijhOalQaTINgyAdOkiFgVtQA59WxDIENh34aR-S0QZQB5q3JcS0j7fJuXlstZ3XfFB0KkDrCURoEr2lAPNE7zKd+gjhhDQoZhOG1SwMAIRwGRC7Fc6IfP8brPMgiAA), seem to report the expected errors:

```
t.setTransitionInProgress(false)
t.setTransitionInProgress(true, new Date())
t.setTransitionInProgress(true, 1701439114)
t.setTransitionInProgress(true, "1970-01-20T16:37:19.114Z")

t.setTransitionInProgress()        // Error: Expected 1-2 arguments, but got 0
t.setTransitionInProgress(true)    // Error: Argument of type 'true' is not assignable to parameter of type 'false'.

```

Reply:
My bad :)